File: naTanjir/UI/natanjir_desktop/lib/models/restoran.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:natanjir_desktop/models/vrsta_restorana.dart';
part 'restoran.g.dart';

@JsonSerializable()
class Restoran {
  int? restoranId;
  String? naziv;
  String? radnoVrijemeOd;
  String? radnoVrijemeDo;
  String? slika;
  String? lokacija;
  int? vrstaRestoranaId;
  int? vlasnikId;
  bool? isDeleted;
  VrstaRestorana? vrstaRestorana;
  Restoran();
  factory Restoran.fromJson(Map<String, dynamic> json) =>
      _$RestoranFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$RestoranToJson(this);
}

File: naTanjir/naTanjir.Model/Restoran.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace naTanjir.Model
{
    public class Restoran
    {
        public int RestoranId { get; set; }

        public string Naziv { get; set; } = null!;

        public string RadnoVrijemeOd { get; set; } = null!;

        public string RadnoVrijemeDo { get; set; } = null!;

        public byte[]? Slika { get; set; }

        public string Lokacija { get; set; } = null!;

        public int VrstaRestoranaId { get; set; }
        public bool? IsDeleted { get; set; }

        public int VlasnikId { get; set; }

        public virtual VrstaRestorana VrstaRestorana { get; set; } 
    }
}
